python/pipeline/pupil.py

```diff
@@ -596,27 +596,75 @@ class Tracking(dj.Computed):
+        roi=NULL                : longblob  # roi of eye
+        gauss_blur=NULL         : float     # bluring of ROI
+        exponent=NULL           : tinyint   # exponent for contrast enhancement
+        dilation_iter=NULL      : tinyint   # number of dilation and erosion operations
+        min_contour_len=NULL    : tinyint   # minimal contour length
+        running_avg_mix=NULL    : float     # weight a in a * current_frame + (1-a) * running_avg
+        """
+
     class ManualTracking(dj.Part):
         definition="""
         -> master
         frame_id                    : int                   # frame id (starting from 0)
         ---
         contour=NULL                : longblob              # eye contour relative to ROI
         """
-        def make(self, key):
-            print("""
-            This method is deprecated!
-            Simply re-inserting data that were manually tracked before (ManuallyTrackedContours)!
-            If data that matches with key does not exist, it will not be stored!
-            """)
-
-            if (ManuallyTrackedContours() & key).fetch1() is not None:
-                for frame_id in range((ManuallyTrackedContours.Frame & key).fetch('frame_id').max()):
-                    ckey = (ManuallyTrackedContours.Frame & dict(key, frame_id=frame_id)).fetch1()
-                    self.insert1(dict(ckey, tracking_method= key['tracking_method']))
+        def make(self, key, backup_file=None):
+            
+            # key does exist in ManuallyTrackedContours (i.e. we tracked before)
+            if len(ManuallyTrackedContours() & key):
+
+                print("""
+                The given key has been tracked manually before (from ManuallyTrackedContours)! 
+                Simply re-inserting previously tracked data here!
+                """)
+                for frame_id in range(len(ManuallyTrackedContours.Frame & key)):
+                    # copy Frame info
+                    frame_key = (ManuallyTrackedContours.Frame & dict(key, frame_id=frame_id)).fetch1()
+                    self.insert1(dict(frame_key, tracking_method= key['tracking_method']))
+                    # copy Parameter info
+                    param_key = (ManuallyTrackedContours.Parameter & dict(key, frame_id=frame_id)).fetch1()
+                    min_lambda = (ManuallyTrackedContours & key).fetch1('min_lambda')
+                    Tracking.ManualTrackingParameter.insert1(
+                        dict(param_key, min_lambda=min_lambda, tracking_method= key['tracking_method']))
+            
+            # key does not exist in ManuallyTrackedContours, hence need to trace manually
             else:
-                print("Given key does not exist in ManuallyTrackedContours table!")
-                print("Either manually track by populating ManuallyTrackedContours or use deeplabcut method!")
+                print("Manually Tracking!")
+            
+                if backup_file is None:
```

Comment:
When you start tracking for the first time, there is no backup_file. If somehow, the the tracking process is crashed in the middle, then one can feed the backup_file that is saved at `/temp` into it and it picks up the process from there.

Reply:
but how will this be feed via populate? or will you call make with the backup file, I would ask taliah or Paul to seee if they ever use that, otherwise we could just drop the backup thing. From what I know if it crashes in the middle they just restart the entire thing

Reply:
I will talk to taliah to figure this out in detail tomorrow. But i see your point